src/storage/index/in_mem_hash_index.cpp

```diff
@@ -155,21 +155,22 @@ bool InMemHashIndex<T>::appendInternal(Key key, common::offset_t value, common::
     auto slotID = HashIndexUtils::getPrimarySlotIdForHash(this->indexHeader, hash);
     SlotIterator iter(slotID, this);
     do {
```

Comment:
By looking at `appendInternal` and `lookup` now, I feel you should be able to refactor the `do while` loop into a single function that can be shared inside both `appendInternal` and `lookup`?

Reply:
Yes, that's probably a good idea. I've done so.